// AUTOMATICALLY GENERATED BY RCPPDEEPSTATE PLEASE DO NOT EDIT BY HAND, INSTEAD EDIT
// fill_reads_DeepState_TestHarness_generation.cpp and fill_reads_DeepState_TestHarness_checks.cpp

#include <fstream>
#include <RInside.h>
#include <iostream>
#include <RcppDeepState.h>
#include <qs.h>
#include <DeepState.hpp>

NumericVector fill_reads(NumericVector new_reads, NumericVector new_counts);

TEST(immunarch_deepstate_test,fill_reads_test){
  RInside R;
  std::cout << "input starts" << std::endl;
  NumericVector new_reads  = RcppDeepState_NumericVector();
  qs::c_qsave(new_reads,"/home/akhila/fuzzer_packages/fuzzedpackages/immunarch/inst/testfiles/fill_reads/inputs/new_reads.qs",
		"high", "zstd", 1, 15, true, 1);
  std::cout << "new_reads values: "<< new_reads << std::endl;
  NumericVector new_counts  = RcppDeepState_NumericVector();
  qs::c_qsave(new_counts,"/home/akhila/fuzzer_packages/fuzzedpackages/immunarch/inst/testfiles/fill_reads/inputs/new_counts.qs",
		"high", "zstd", 1, 15, true, 1);
  std::cout << "new_counts values: "<< new_counts << std::endl;
  std::cout << "input ends" << std::endl;
  try{
    fill_reads(new_reads,new_counts);
  }
  catch(Rcpp::exception& e){
    std::cout<<"Exception Handled"<<std::endl;
  }
}
